penguin/tools/runtime.py

```
@@ -461,6 +464,88 @@ def tool_call_from_responses_info(tool_info: dict[str, Any]) -> Optional[ToolCal
     )
 
 
+def _format_browser_tool_output(action_result: dict[str, Any]) -> str:
```

Ruff (0.15.12)

[warning] Too many branches (20 > 12)

(PLR0912)